<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN"
  "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
<head>
<meta name="generator" content="jemdoc, see http://jemdoc.jaboc.net/" />
<meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<link rel="stylesheet" href="jemdoc.css" type="text/css" />
<title>Publication</title>
</head>
<body>
<table summary="Table for page layout." id="tlayout">
<tr valign="top">
<td id="layout-menu">
<div class="menu-category">Menu</div>
<div class="menu-item"><a href="index.html">Home</a></div>
<div class="menu-item"><a href="publication.html" class="current">Publications</a></div>
<div class="menu-item"><a href="teaching.html">CS590&nbsp;(Graphs&nbsp;in&nbsp;Machine&nbsp;Learning)</a></div>
<div class="menu-item"><a href="teaching_ba.html">CS590&nbsp;(Bioinformatics&nbsp;Algorithms)</a></div>
<div class="menu-item"><a href="software.html">Software</a></div>
</td>
<td id="layout-content">
<div id="toptitle">
<h1></h1>
<h1></h1>

<body><div class="container">
<div class="floatRight">
<table>
	<tbody>
		<tr valign="top">
			<td id="layout-content">
			<div id="toptitle">
			<h2>Machine Learning Papers</h2>
			&nbsp;
			<ul>
	
	<li> <p><font color=red size=+0.5<b><i>NEW!</i> </b></font> <font color=DarkBlue size=+0.5><b>DecompDiff: Diffusion Models with Decomposed Priors for Structure-Based Drug Design
</b></font><br />
        Jiaqi Guan, Xiangxin Zhou, Yuwei Yang, Yu Bao, Jian Peng, <b>Jianzhu Ma</b>, Qiang Liu, Liang Wang, Quanquan Gu <br />
        ICML 2023 <br /></p>
        &nbsp;
        </li>
				
	<li> <p><font color=red size=+0.5<b><i>NEW!</i> </b></font> <font color=DarkBlue size=+0.5><b>Addressing the Atom-Bond Inconsistency Problem in 3D Molecule Generation
</b></font><br />
        Xingang Peng, Jiaqi Guan, Qiang Liu, <b>Jianzhu Ma</b> <br />
        ICML 2023 <br /></p>
        &nbsp;
        </li>
				
	<li> <p><font color=red size=+0.5<b><i>NEW!</i> </b></font> <font color=DarkBlue size=+0.5><b>Rotamer Density Estimators are Unsupervised Learners of the Effect of Mutations on Protein-Protein Interaction
</b></font><br />
        Shitong Luo, Yufeng Su, Zuofan Wu, Chenpeng Su, Jian Peng, <b>Jianzhu Ma</b> <br />
        ICLR 2023 <br /></p>
        &nbsp;
        </li>
	
	<li> <p><font color=red size=+0.5<b><i>NEW!</i> </b></font> <font color=DarkBlue size=+0.5><b>Learning Sparse Group Models Through Boolean Relaxation</b></font><br />
        Yijie Wang, Yuan Zhou, Xiaoqing Huang, Kun Huang, Jie Zhang, <b>Jianzhu Ma</b> <br />
        ICLR 2023 <br /></p>
        &nbsp;
        </li>
	
	<li> <p><font color=red size=+0.5<b><i>NEW!</i> </b></font> <font color=DarkBlue size=+0.5><b>Boosting the Cycle Counting Power of Graph Neural Networks with I^2-GNNs
</b></font><br />
        Yinan Huang, Xingang Peng, Jianzhu Ma, <b>Jianzhu Ma</b>, Muhan Zhang <br />
        ICLR 2023 <br /></p>
        &nbsp;
        </li>
	
	<li> <p><font color=red size=+0.5<b><i>NEW!</i> </b></font> <font color=DarkBlue size=+0.5><b>3D Equivariant Diffusion for Target-Aware Molecule Generation and Affinity Prediction</b></font><br />
        Jiaqi Guan, Wesley Wei Qian, Xingang Peng, Yufeng Su, Jian Peng, <b>Jianzhu Ma</b> <br />
        ICLR 2023 <br /></p>
        &nbsp;
        </li>

	<li> <p> <font color=DarkBlue size=+0.5><b>Device-System End-to-End Design of Photonic Neuromorphic Processor Using Reinforcement Learning
</b></font><br />
        Yingheng Tang, Princess Tara Zamani, Ruiyang Chen, <b>Jianzhu Ma</b>, Minghao Qi, Cunxi Yu, Weilu Gao <br />
        Laser & Photonics reviews 2022 <br /></p>
        &nbsp;
        </li>	
				
	<li> <p> <font color=DarkBlue size=+0.5><b>Antigen-Specific Antibody Design and Optimization with Diffusion-Based Generative Models</b></font><br />
        Shitong Luo, Yufeng Su, Xingang Peng, Sheng Wang, Jian Peng, <b>Jianzhu Ma</b> <br />
        NeurlPS 2022 <br /></p>
        &nbsp;
        </li>	
				
	<li> <p> <font color=DarkBlue size=+0.5><b>Efficient Meta Reinforcement Learning for Preference-based Fast Adaptation</b></font><br />
        Zhizhou Ren, Anji Liu, Yitao Liang, Jian Peng, <b>Jianzhu Ma</b> <br />
        NeurlPS 2022 <br /></p>
        &nbsp;
        </li>	
				
	<li> <p> <font color=DarkBlue size=+0.5><b> MobilePrune: Neural Network Compression via â„“0 Sparse Group Lasso on 
the Mobile System
</b></font><br />
        Yubo Shao, Kaikai Zhao, Zhiwen Cao, Zhehao Peng, Xingang Peng, Pan Li, Yijie Wang, <b> Jianzhu Ma </b> <br />
        Sensor 2022 <br /></p>
        &nbsp;
        </li>
				
	
	<li> <p> <font color=DarkBlue size=+0.5><b> Multiwave COVID-19 Prediction from Social Awareness using Web Search and Mobility Data
</b></font><br />
        Jiawei Xue, Takahiro Yabe, Kota Tsubouchi, <b> Jianzhu Ma </b>, Satish V. Ukkusuri <br />
        KDD 2022 <font color=red <b><i>corresponding author</i> </b></font> <font color=red <b><i>Oral presentation</i> </b></font> <br /></p>
        &nbsp;
        </li>
				
	<li> <p> <font color=DarkBlue size=+0.5><b> Proximal Exploration for Model-guided Biological Sequence Design
</b></font><br />
        Zhizhou Ren, Jiahan Li, Fan Ding, Yuan Zhou, <b> Jianzhu Ma </b>, Jian Peng <br />
        ICML 2022 <font color=red <b><i>Short presentation</i> </b></font> <br /></p>
        &nbsp;
        </li>
	
	<li> <p> <font color=DarkBlue size=+0.5><b> Pocket2Mol: Efficient Molecular Sampling Based on 3D Protein Pockets
</b></font><br />
        Xingang Peng, Shitong Luo, Jiaqi Guan, Qi Xie, Jian Peng, <b> Jianzhu Ma </b> <br />
        ICML 2022 <font color=red <b><i>Short presentation</i> </b></font> <br /></p>
        &nbsp;
        </li>
	
	<li> <p> <font color=DarkBlue size=+0.5><b> 3DLinker: An E(3) Equivariant Variational Autoencoder for Molecular Linker Design
</b></font><br />
        Yinan Huang, Xingang Peng, <b> Jianzhu Ma </b>, Muhan Zhang <br />
        ICML 2022 <font color=red <b><i>corresponding author</i> </b></font> <font color=red <b><i>Long presentation</i> </b></font> <br /></p>
        &nbsp;
        </li>
				
	<li> <p> <font color=DarkBlue size=+0.5><b> Equivariant Point Set Analysis via Learning Orientations for Message Passing
</b></font><br />
        Shitong Luo, Jiahan Li, Jiaqi Guan, Yufeng Su, Chaoran Cheng, Jian Peng, <b> Jianzhu Ma </b> <br />
        CVPR 2022 <font color=red <b><i>Oral presentation</i> </b></font> <br /></p>
        &nbsp;
        </li>
				
	<li> <p>  <font color=DarkBlue size=+0.5><b> Physics-Guided and Physics-Explainable Recurrent Neural Network for Time Dynamics in Optical Resonances
</b></font><br />
        Yingheng Tang, Jichao Fan, Xinwei Li,  <b> Jianzhu Ma </b>, Minghao Qi, Cunxi Yu, Weilu Gao <br />
        Nature Computational Science 2022 <br /></p>
        &nbsp;
        </li>	
				
	<li> <p> <font color=DarkBlue size=+0.5><b> Neural Energy Minimization for Molecular Conformation Optimization
</b></font><br />
        Jiaqi Guan, Wei Qian, Qiang Liu, Wei-ying Ma, <b>Jianzhu Ma</b>, Jian Peng <br />
        ICLR 2022 <br /></p>
        &nbsp;
        </li>			
				
	<li> <p> <font color=DarkBlue size=+0.5><b>Neural Predicting Higher-order Patterns in Temporal Networks
</b></font><br />
        Yuyun Liu, <b>Jianzhu Ma</b>, Pan Li <br />
        WWW 2022<br /></p>
        &nbsp;
        </li>
				
        <li> <p> <font color=DarkBlue size=+0.5><b>Quantifying spatial homogeneity of urban road networks via graph neural networks</b></font><br />
        Jiawei Xue, Nan Jiang, Senwei Liang, Qiyuan Pang, Takahiro Yabe, Satish Ukkusuri, <b>Jianzhu Ma</b><br />
        Nature Machine intelligence 2022 <font color=red <b><i>cover paper</i> </b></font> <br /></p>
        </li>

	<ul>
        <li><p><font color=red size=+0.5<b>TechExplore:</b></font> <a href="https://techxplore.com/news/2022-05-graph-neural-networks-spatial-homogeneity.html">Using graph neural networks to measure the spatial homogeneity of road networks
</a><br /></p>
        </li>
	<li><p><font color=red size=+0.5<b>Nature Computational Science:</b></font> <a href="https://www.nature.com/articles/s43588-022-00244-x">Gauging urban development with neural networks
</a><br /></p>
        </li>
        </ul>

        <li> <p> <font color=DarkBlue size=+0.5><b>A 3D Generative Model for Structure-Based Drug Design</b></font><br />
        Shitong Luo, Jiaqi Guan, <b>Jianzhu Ma</b>, Jian Peng <br />
        NeurlPS 2021 <br /></p>
        &nbsp;
        </li>

        <li> <p> <font color=DarkBlue size=+0.5><b> Fast Projection onto the Capped Simplex withApplications to Sparse Regression in Bioinformatics </b></font><br />
        Andersen Ang, <b>Jianzhu Ma</b>, Nianjun Liu, Kun Huang, Yijie Wang <br />
        NeurlPS 2021 <br /></p>
        &nbsp;
        </li>

        <li> <p> <font color=DarkBlue size=+0.5><b> XOR-CD: Linearly Convergent Constrained Structure Generation </b></font><br />
        Fan Ding, <b>Jianzhu Ma</b>, Jinbo Xu, Yexiang Xue<br />
        ICML 2021 <font color=red <b><i>oral presentation</i> </b></font></p>
        &nbsp;
        </li>

        <li> <p> <font color=DarkBlue size=+0.5><b>AirSign: Smartphone Authentication by Signing in the Air</b></font><br />
        Yubo Shao, Tinghan Yang, He Wang, <b>Jianzhu Ma</b><br />
        Sensor 2021<br /></p>
        &nbsp;
        </li>

        <li> <p> <font color=DarkBlue size=+0.5><b> PALM: Probabilistic Area Loss Minimization for Protein Sequence Alignment </b></font><br />
        Fan Ding, Nan Jiang, <b>Jianzhu Ma</b>, Jian Peng, Jinbo Xu, Yexiang Xue<br />
        UAI 2021</p>
        &nbsp;
        </li>

          <li>
          <p> <font color=DarkBlue size=+0.5><b> Breaking the Limit of Graph Neural Networks by Improving the Assortativity of Graphs with Local Mixing Patterns </b></font><br />
          Susheel Suresh, Vinith Budde, Jennifer Neville, Pan Li, <b>Jianzhu Ma</b><br />
          KDD 2021<br /></p>
          &nbsp;
          </li>

          <li>
          <p><font color=DarkBlue size=+0.5><b>Causality in machine learning: When causal inference meets deep neural networks</b></font><br />
          Yunan Luo, Jian Peng, <b>Jianzhu Ma</b> <br />
          Nature Machine intelligence 2020<br /></p>
          &nbsp;
          </li>

          <li>
          <p><font color=DarkBlue size=+0.5><b>Structure Learning Constrained by Node-Specific Degree Distribution</b></font><br />
          <b>Jianzhu Ma</b>, Sheng Wang and Jinbo Xu<br />
          UAI 2015<br /></p>
          &nbsp;
          </li>

          <li>
          <p><font color=DarkBlue size=+0.5><b>Estimating the Partition Function of Graphical Models Using Langevin Importance Sampling</b></font><br />
          <b>Jianzhu Ma</b>, Jian Peng, Sheng Wang and Jinbo Xu<br />
          AI &amp; Statistics 2013 <font color=red <b><i>oral presentation</i> </b></font> <br /></p>
          &nbsp;
          </li>

			  </ul>
			</div>
			</td>
		</tr>
	</tbody>
</table>
</div>

<div class="floatLeft">
<table>
	<tbody>
		<tr valign="top">
			<td id="layout-content">
			<div id="toptitle">
			<h2>Biology Papers</h2>
			&nbsp;
			<ul>
				
	<li> <p> <font color=red size=+0.5<b><i>NEW!</i> </b></font> <font color=DarkBlue size=+0.5><b> Sagittarius: Extrapolating Heterogeneous Time-Series Gene Expression Data
 </b></font><br />
        Addie Woicik, Mingxin Zhang, Janelle Chan, <b>Jianzhu Ma</b>, Sheng Wang <br />
        Nature Machine intelligence 2023<br /></p>
        &nbsp;
        </li>
				
	<li> <p> <font color=red size=+0.5<b><i>NEW!</i> </b></font> <font color=DarkBlue size=+0.5><b> Con-AAE: Contrastive Cycle Adversarial Autoencoders for Single-cell Multi-omics Alignment and Integration
 </b></font><br />
        Xuesong Wang, Zhihang Hu, Tingyang Yu, Yixuan Wang, Ruijie Wang, Yumeng Wei, Juan Shu, <b> Jianzhu Ma </b> and Yu Li <br />
        Bioinformatics 2023<br /></p>
        &nbsp;
        </li>
				
				
	<li> <p> <font color=red size=+0.5<b><i>NEW!</i> </b></font> <font color=DarkBlue size=+0.5><b> Characterizing the interaction conformation between T cell receptors and epitopes with deep learning
 </b></font><br />
        Xingang Peng, Yipin Lei, Peiyuan Feng, Lemei Jia, <b> Jianzhu Ma </b>, Dan Zhao and Jianyang Zeng <br />
        Nature Machine intelligence 2023<br /></p>
        &nbsp;
        </li>	
				
				
	<li> <p> <font color=DarkBlue size=+0.5><b> Pisces: A combo-wise contrastive learning approach to synergistic drug combination prediction
 </b></font><br />
        Jiacheng Lin, Hanwen Xu, Addie Woicik, <b> Jianzhu Ma </b> and Sheng Wang <br />
        RECOMB 2023<br /></p>
        &nbsp;
        </li>			
				
	<li> <p> <font color=DarkBlue size=+0.5><b> Boosting single-cell gene regulatory network
reconstruction via bulk-cell transcriptomic data
 </b></font><br />
        Hantao Shu, Fan Ding, Jingtian Zhou, Yexiang Xue, Dan Zhao, Jianyang Zeng and <b> Jianzhu Ma </b> <br />
        Briefings in Bioinformatics 2022<br /></p>
        &nbsp;
        </li>	
				
	<li> <p> <font color=DarkBlue size=+0.5><b>Single-cell multiomics analysis reveals regulatory programs in clear cell renal cell carcinoma
 </b></font><br />
        Qi Xie, Zhilin Long, Chengfang Sun, Min Tang, Yin Wang, Jiayan Ma, Jichuan Yu, Jingchao Wei, <b> Jianzhu Ma </b>, Bohan Wang, and Jiaming Wen <br />
        Cell Discovery 2022<br /></p>
        &nbsp;
        </li>			
				
	<li> <p> <font color=DarkBlue size=+0.5><b>Deep learning guided optimization of human antibody against SARS-Cov-2 variants with broad neutralization </b></font><br />
        Sisi Shan, Shitong Luo, Ziqing Yang, Junxian Hong, Yufeng Su, Fan Ding, Chengyu Li, Peng Chen, <b>Jianzhu Ma</b>, Xuanling Shi, Qi Zhang, Bonnie Berger, Linqi Zhang, Jian Peng<br />
        PNAS 2022<br /></p>
        &nbsp;
        </li>
				
	<li> <p> <font color=DarkBlue size=+0.5><b>scPretrain: Multi-task self-supervised learning for cell type classification</b></font><br />
        Ruiyi Zhang, Yunan Luo, <b>Jianzhu Ma</b>, Ming Zhang, Sheng Wang <br />
        Bioinformatics 2022<br /></p>
        &nbsp;
        </li>
				
	<li> <p> <font color=DarkBlue size=+0.5><b>Next decadeâ€™s AI-based drug development features the tight integration of data and computation</b></font><br />
        Yunan Luo, Jian Peng, <b>Jianzhu Ma</b> <br />
        Health Data Science 2021<br /></p>
        &nbsp;
        </li>
				
				
        <li> <p> <font color=DarkBlue size=+0.5><b>A multi-scale map of cell structure fusing protein images and interactions</b></font><br />
        Yue Qin, Edward L. Huttlin, Casper F. Winsnes, Maya L. Gosztyla, Ludivine Wacheul, Marcus R. Kelly, Steven M. Blue, Fan Zheng, Michael Chen, Leah V. Schaffer, Katherine Licon, Anna BÃ¤ckstrÃ¶m,
        Laura Pontano Vaites, John J. Lee, Wei Ouyang, Sophie N. Liu, Tian Zhang, Erica Silva, Jisoo Park, Adriana Pitea, Jason F. Kreisberg, Steven P. Gygi, <b>Jianzhu Ma</b>, J. Wade Harper, Gene W. Yeo,
        Denis L. J. Lafontaine, Emma Lundbergï¼ŒTrey Ideker<br />
        Nature 2021<br /></p>
        &nbsp;
        </li>

        <li><p> <font color=DarkBlue size=+0.5><b> SARS-CoV-2 exacerbates proinflammatory responses in myeloid cells through ACE2-independent interactions via C-type lectin receptors and TTYH2 </b></font><br />
        Qiao Lu, Jia Liu, Shuai Zhao, Maria Florencia Gomez Castro, Maudry Laurent-Rolle, Jianbo Dong, Xiaojuan Ran, Payal Damani-Yokota, Hongzheng Tang, Triantafyllia Karakousi, Juhee Son, Maria E. Kaczmarek, Ze Zhang, Stephen T. Yeung, Broc T. McCune, Rita E. Chen, Fei Tang, Xianwen Ren, Xufeng Chen, Jack C.C. Hsu, Marianna Teplova, Betty Huang, Haijing Deng, Zhilin Long, Tenny Mudianto, Shumin Jin, Peng Lin, Jasper Du, Ruochen Zang, Tina Tianjiao Su, Alberto Herrera, Ming Zhou, Renhong Yan, Jia Cui, James Zhu, Qiang Zhou, Tao Wang, <b>Jianzhu Ma</b>, Sergei B. Koralov, Zemin Zhang, Iannis Aifantis, Leopoldo N. Segal, Michael S. Diamond, Kamal M. Khanna, Kenneth A. Stapleford, Peter Cresswel, Yue Liu, Siyuan Ding, Qi Xie, Jun Wang<br />
        Immunity 2021 (Impact Factor=22)<br /></p>
        &nbsp;
        </li>

        <li><p> <font color=DarkBlue size=+0.5><b> Modeling Gene Regulatory Networks Using Neural Network Architectures </b></font><br />
        Hantao Shu, Jingtian Zhou, Qiuyu Lian, Han Li, Dan Zhao, Jianyang Zeng, <b>Jianzhu Ma</b><br />
        Nature Computational Science 2021 <br /></p>
        &nbsp;
        </li>

        <li><p> <font color=DarkBlue size=+0.5><b> ParsVNN: parsimony visible neural networks for uncovering cancer-specific and drug-sensitive genes and pathways </b></font><br />
        Xiaoqing Huang, Kun Huang, Travis Johnson, Milan Radovich, Jie Zhang, <b>Jianzhu Ma</b>, Yijie Wang<br />
        NAR genomics and bioinformatics 2021 <font color=red <b><i>corresponding author</i> </b></font><br /></p>
        &nbsp;
        </li>

        <li><p> <font color=DarkBlue size=+0.5><b>Predicting MHC-peptide binding affinity by differential boundary tree</b></font><br />
        Peiyuan Feng, Jianyang Zeng and <b>Jianzhu Ma</b><br />
        ISMB 2021<br /></p>
        &nbsp;
        </li>

        <li><p> <font color=DarkBlue size=+0.5><b>Disease Gene Prediction with Privileged Information and Heteroscedastic Dropout</b></font><br />
        Juan Shu, Yu Li, Sheng Wang, Bowei Xi and <b>Jianzhu Ma</b><br />
        ISMB 2021<br /></p>
        &nbsp;
        </li>

        <li><p> <font color=DarkBlue size=+0.5><b>Towards the prediction of higher-order genetic interactions</b></font><br />
        Ruochi Zhang, <b>Jianzhu Ma</b>, Jian Ma<br />
        RECOMB 2021 <font color=red <b><i>corresponding author</i> </b></font><br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Predicting drug response and synergy using a deep learning model of a cancer cell</b></font><br />
        Brent Kuenzi, Jisoo Park, Samson H. Fong, Kyle Salinas Sanchez, Jason F. Kreisberg, <b>Jianzhu Ma</b>, Trey Ideker<br />
        Cancer Cell 2020<br /></p>
        </li>

        <ul>
        <li><p><font color=red size=+0.5<b>ScienceDaily:</b></font> <a href="https://www.sciencedaily.com/releases/2020/10/201026164900.htm">DrugCell: New experimental AI platform matches tumor to best drug combo</a><br /></p>
        </li>
        <li><p><font color=red size=+0.5<b>Health IT Analytics:</b></font> <a href="https://healthitanalytics.com/news/deep-learning-model-could-enhance-cancer-precision-medicine">Deep Learning Model Could Enhance Cancer Precision Medicine</a><br /></p>
        </li>
        <li><p><font color=red size=+0.5<b>Med India:</b></font> <a href="https://www.medindia.net/news/new-ai-platform-matches-tumor-to-best-drug-combination-198154-1.htm">New AI Platform Matches Tumor to Right Drug Combination</a></p>
        </li>
        <li><p><font color=red size=+0.5<b>AZO Life Sciences:</b></font> <a href="https://www.azolifesciences.com/news/20201026/Experimental-AI-platform-predicts-best-approach-to-cancer-treatment.aspx">Experimental AI platform predicts best approach to cancer treatment</a></p>
        </li>
        <li><p><font color=red size=+0.5<b>UC San Diego Health Newsroom:</b></font> <a href="https://health.ucsd.edu/news/releases/Pages/2020-10-22-drugcell-new-experimental-ai-platform-matches-tumor-to-best-drug-combo.aspx">DrugCell: New Experimental AI Platform Matches Tumor to Best Drug Combo</a></p>
        </li>
        </ul>

        <li><p><font color=DarkBlue size=+0.5><b>A conserved epigenetic progression aligns dog and human age</b></font><br />
        Tina Wang, <b>Jianzhu Ma</b>, Andrew N. Hogan, Samson Fong, Katherine Licon, Brian Tsui, Jason F. Kreisberg, Peter D. Adams, Anne-Ruxandra Carvunis, Danika L. Bannasch, Elaine A. Ostrander and Trey Ideker<br />
        Cell Systems 2020<br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Artificial-Cell-Type Aware Cell Type Classification in CITE-seq</b></font><br />
        Qiuyu Lian, Hongyi Xin, <b>Jianzhu Ma</b>, Liza Konnikova, Wei Chen, Jin Gu and Kong Chen<br />
        ISMB 2020<br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Learning predictive models of drug response that translate across biological contexts</b></font><br />
        <b>Jianzhu Ma</b>, Samson Fong, Yunan Luo, Christopher Bakkenist, John Paul Shen, Soufiane Mourragui, Lodewyk Wessels, Marc Hafner, Roded Sharan, Jian Peng, Trey Ideker<br />
        Nature Cancer 2020<br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Robust Single-Cell Hi-C Clustering by Convolution and Random Walk based Imputation</b></font><br />
        Jingtian Zhou, <b>Jianzhu Ma</b>, Yusi Chen, Chuankai Cheng, Bokan Bao, Jian Peng, Terrence J. Sejnowski, Jesse R. Dixon, Joseph R. Ecker<br />
        PNAS 2019 <font color=red <b><i>co-first author</i> </b></font><br /></p>
        <ul>
        <li><p><font color=red size=+0.5<b>SALK NEWS:</b></font> <a href="https://www.salk.edu/news-release/new-computational-tool-lets-researchers-identify-cells-based-on-their-chromosome-shape/">NEW COMPUTATIONAL TOOL LETS RESEARCHERS IDENTIFY CELLS BASED ON THEIR CHROMOSOME SHAPE</a><br /></p>
        </li>
        <li><p><font color=red size=+0.5<b>GenomeWeb:</b></font> <a href="https://www.genomeweb.com/scan/pnas-studies-look-lung-cancer-lncrna-hi-c-clustering-and-more">PNAS Studies Look at Lung Cancer LncRNA, Hi-C Clustering, and More</a></p>
        </li></ul>
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>DDOT: A Swiss-Army Knife for Investigating Data-Driven Biological Ontologies</b></font><br />
        Michael Ku Yu, <b>Jianzhu Ma</b>, Keiichiro Ono, Fan Zheng, Samson H. Fong, Aaron Gary, Jing Chen, Barry Demchak, Dexter Pratt, Trey Ideker<br />
        Cell Systems 2019 <font color=red <b><i>co-first author</i> </b></font> <br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Mitigating Data Scarcity in Protein Binding Prediction Using Meta-Learning</b></font><br />
        Yunan Luo, <b>Jianzhu Ma</b>, Xiaoming Zhao, Yufeng Su, Yang Liu, Trey Ideker, Jian Peng<br />
        RECOMB 2019 <font color=red <b><i>co-first author</i> </b></font> <br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Visible Machine Learning for Biomedicine</b></font><br />
        Michael Ku Yu, <b>Jianzhu Ma</b>, Jasmin Fisher, Jason F. Kreisberg, Benjamin J. Raphael, Trey Ideker<br />
        Cell 2018 <font color=red <b><i>co-first author</i> </b></font> <br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Typing tumors using pathways selected by somatic evolution</b></font><br />
        Sheng Wang, <b>Jianzhu Ma</b>, Wei Zhang, John Shen, Justin Huang, Jian Peng, Trey Ideker<br />
        Nature Communications 2018 <font color=red <b><i>co-first author</i> </b></font> <br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Classifying tumors by supervised network propagation</b></font><br />
        Wei Zhang, <b>Jianzhu Ma</b>, Trey Ideker<br />
        ISMB 2018 <font color=red <b><i>co-first author</i> </b></font> <br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Using deep learning to model the hierarchical structure and function of a cell</b></font><br />
        <b>Jianzhu Ma</b>, Michael Ku Yu, Samson Fong, Keiichiro Ono, Eric Sage, Barry Demchak, Roded Sharan, Trey Ideker<br />
        Nature Methods 2018<br /></p>
        <ul>
        <li><p><font color=red size=+0.5<b>Cover Art:</b></font> <a href="https://www.nature.com/nmeth/journal/v15/n4/covers/index.html">Nature Methods About the Cover: April 2018 Vol 15 No 4</a><br /></p>
        </li>
        <li><p><font color=red size=+0.5<b>IEEE SPECTRUM:</b></font> <a href="https://spectrum.ieee.org/the-human-os/biomedical/diagnostics/cracking-open-the-black-box-of-ai-with-cell-biology">Cracking Open the Black Box of AI with Cell Biology</a><br /></p>
        </li>
        <li><p><font color=red size=+0.5<b>UCSD Health:</b></font> <a href="https://health.ucsd.edu/news/releases/Pages/2018-03-05-how-a-yeast-cell-helps-crack-open-the-black-box-behind-artificial-intelligence.aspx">How a Yeast Cell Helps Crack Open the &ldquo;Black Box&rdquo; Behind Artificial Intelligence</a><br /></p>
        </li>
        <li><p><font color=red size=+0.5<b>GenomeWeb:</b></font> <a href="https://www.genomeweb.com/informatics/deep-learning-based-cell-model-may-predict-patient-tumor-cell-growth-someday#.WtUmt4gbOUk">Deep Learning-Based Cell Model May Predict Patient Tumor Cell Growth Someday</a><br /></p>
        </li>
        <li><p><font color=red size=+0.5<b>Nature:</b></font> <a href="https://www.nature.com/articles/d41586-020-01128-8">Deep learning takes on tumours</a><br /></p>
        </li></ul>
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Deciphering signaling specificity with deep neural networks</b></font><br />
        Yunan Luo, <b>Jianzhu Ma</b>, Yang Liu, Qing Ye, Trey Ideker, Jian Peng<br />
        RECOMB 2018 <font color=red <b><i>co-first author</i> </b></font> <br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Annotating gene sets by mining large literature collections with protein networks</b></font><br />
        Sheng Wang, <b>Jianzhu Ma</b>, Michael Ku Yu, Fan Zheng, Edward W Huang, Jiawei Han, Jian Peng, Trey Ideker<br />
        Pacific Symposium on Biocomputing, 2018 <font color=red <b><i>co-first author</i> </b></font> <br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>DeepBound: Accurate Identification of Transcript Boundaries via Deep Convolutional Neural Fields</b></font><br />
        Mingfu Shao, <b>Jianzhu Ma</b>, Sheng Wang<br />
        ISMB 2017 <font color=red <b><i>co-first author</i> </b></font> <br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>AUCpreD: proteome-level protein disorder prediction by AUC-maximized Deep Convolutional Neural Fields</b></font><br />
        Sheng Wang, <b>Jianzhu Ma</b>, Jinbo Xu<br />
        ECCB 2016<br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>ModuleAlign: Module-based global alignment of protein-protein interaction networks</b></font><br />
        Somaye Hashemifar, <b>Jianzhu Ma</b>, Hammad Naveed, Stefan Canzar, Jinbo Xu<br />
        ECCB 2016<br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Protein secondary structure prediction using deep convolutional neural fields</b></font><br />
        Sheng Wang, Jian Peng, <b>Jianzhu Ma</b>, Jinbo Xu<br />
        Scientific Reports 2016<br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Bermuda: de novo assembly of transcripts with new insights for handling uneven coverage</b></font><br />
        Qingming Tang, Sheng Wang, Jian Peng, <b>Jianzhu Ma</b>, Jinbo Xu<br />
        ACM BCB 2016<br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>AcconPred: Predicting Solvent Accessibility and Contact Number Simultaneously by a Multi-Task Learning Framework under the Conditional Random Fields Model</b></font><br />
        <b>Jianzhu Ma</b> and Sheng Wang<br />
        BioMed Research International 2015<br /> </p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Protein Contact Prediction by Integrating Joint Evolutionary Coupling Analysis and Supervised Learning</b></font><br />
        <b>Jianzhu Ma</b>, Sheng Wang, Zhiyong Wang and Jinbo Xu<br />
        RECOMB and Biofinformatics 2015<br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>MRFalign: Protein Homology Detection through Alignment of Markov Random Fields</b></font><br />
        <b>Jianzhu Ma</b>, Sheng Wang, Jinbo Xu<br />
        RECOMB and PLOS Computational Biology 2014<br />
        This work also won the following awards:</p>
        <ul>
        <li><p><font color=red size=+0.5<b>Best Paper Award</b></font> of RECOMB 2014<br /></p>
        </li>
        <li><p><font color=red size=+0.5<b>Best Poster Award</b></font> of 2nd Zing Conference on Protein and RNA Structure Prediction<br /></p>
        </li>
        <li><p><font color=red size=+0.5<b>Warren DeLano Award</b></font> of 3DSIG 2014<br /></p>
        </li></ul>
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Predicting diverse M-best protein contact maps</b></font><br />
        Siqi Sun,<b>Jianzhu Ma</b>, Jinbo Xu<br />
        BIBM 2015<br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>DeepCNF-D: Predicting Protein Order/Disorder Regions by Weighted Deep Convolutional Neural Fields</b></font><br />
        Sheng Wang, <b>Jianzhu Ma</b>, Jinbo Xu<br />
        International Journal of Molecular Sciences 2015<br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>An Integrative Computational Approach for Prioritization of Genomic Variants</b></font><br />
        Inna Dubchak, Sandhya Balasubramanian, Sheng Wang, Cem Meyden, Dinanath Sulakhe, Alexander Poliakov, Daniela BÃ¶rnigen, Bingqing Xie, Andrew Taylor, <b>Jianzhu Ma</b>, Alex R Paciorkowski, Ghayda M Mirzaa, Paul Dave, Gady Agam, Jinbo Xu, Lihadh Al-Gazali, Christopher E Mason, M Elizabeth Ross, Natalia Maltsev and T Conrad Gilliam<br />
        PLOS ONE 2015<br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Algorithms, Applications and Challenges of Protein Structure Alignment</b></font><br />
        <b>Jianzhu Ma</b> and Sheng Wang<br />
        Journal of Advances in Protein Chemistry and Structural Biology 2014<br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Protein Threading Using Context-Specific Alignment Potential</b></font><br />
        <b>Jianzhu Ma</b>, Sheng Wang and Jinbo Xu<br />
        ISMB 2013<br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>A Conditional Neural Fields Model for Protein Threading</b></font><br />
        <b>Jianzhu Ma</b>, Jian Peng, Sheng Wang and Jinbo Xu<br />
        ISMB 2012<br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Fast and Automatic Pairwise Structure Alignment Algorithm close to Manually-Curated Precision</b></font><br />
        Scientific Reports and 3DSIG 2012. <font color=red size=+0.5><b>Warren DeLano Award</b></font><br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>A Self-Adaptive Mechanism for Software Configuration Based on Case-based Reasoning and Policy</b></font><br />
        Maoguang Wang, Hong Mei, Wenpin Jiao, Junjing Jie and <b>Jianzhu Ma</b><br />
        AICI 2010<br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>An Empirical Study on the Impacts of Autonomy of Components on Qualities of Software Systems</b></font><br />
        Shiting Xun, <b>Jianzhu Ma</b>, Wenpin Jiao and Fanjing Meng<br />
        EASe 2011<br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>A Two-way Selections based Approach to Automated Integration of Multi-Agent Systems</b></font><br />
        Xi Fang, <b>Jianzhu Ma</b>, Maoguang Wang and Wenpin Jiao<br />
        3rd National Conference on Agent Theory and Application 2010<br /></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>A Browser-based Framework for Data Cache in Web-Delivered Service Composition</b></font><br />
        Jiyu Huang, Xuanzhe Liu, Qi Zhao, <b>Jianzhu Ma</b> and Gang Huang<br />
        SOCA 2010 <font color=red size=+0.5><b>(Best paper running up)</b></font></p>
        &nbsp;
        </li>

        <li><p><font color=DarkBlue size=+0.5><b>Design and Implementation of a Cooperate Framework for Autonomous Component based on Rule Injection</b></font><br />
        <b>Jianzhu Ma</b>, Jiyu Huang and Wenpin Jiao<br />
        NASAC 2009<br /></p>
        &nbsp;
        </li>

			</ul>
			</div>
			</td>
		</tr>
	</tbody>
</table>
</div>
</div>


</td>
</tr>
</table>
</body>
</html>
