<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN"
  "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
<head>
<meta name="generator" content="jemdoc, see http://jemdoc.jaboc.net/" />
<meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<link rel="stylesheet" href="jemdoc.css" type="text/css" />
<link rel="stylesheet" href="person.css" type="text/css" />
<title></title>
</head>
<body>
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
try {
    var pageTracker = _gat._getTracker("UA-31107207-1");
    pageTracker._trackPageview();
} catch(err) {}</script>
<table summary="Table for page layout." id="tlayout">
<tr valign="top">
<td id="layout-menu">
<div class="menu-category">Menu</div>
<div class="menu-item"><a href="index.html" class="current">Home</a></div>
<div class="menu-item"><a href="publication.html">Publications</a></div>
<div class="menu-item"><a href="teaching.html">CS590&nbsp;(Graphs&nbsp;in&nbsp;Machine&nbsp;Learning)</a></div>
<div class="menu-item"><a href="teaching_ba.html">CS590&nbsp;(Bioinformatics&nbsp;Algorithms)</a></div>
<div class="menu-item"><a href="software.html">Software</a></div>
</td>
<td id="layout-content">
<h1>00230212: Intelligent Bio-computation (智能生物计算) (Fall 2024)</h1>
<h2>Course Information</h2>
<p>When: Monday 19:20 pm &ndash; 20:55 pm. <br />
Where: 清华大学法律图书馆B112. <br />
Instructor: Jianzhu Ma (马剑竹)
Email: majianzhu at tsinghua dot edu dot cn.<br />
Office Hour: Tuesday. 19:30 pm &ndash; 20:30 pm.or by appointment (send e-mail) Where: TBD<br />

</p>
<h2>Course description:</h2>
<p>This course encompasses all the principal areas within bioinformatics and computational biology, making it an 'all-you-need&rsquo; program for those aspiring to craft algorithms for analyzing and interpreting biological data. Students who complete this course will gain insights into a range of fundamental algorithmic strategies, drawn directly from seminal and highly referenced recent literature. The course imparts foundational knowledge in statistics, mathematics, and machine learning, including deep learning aspects. Both lectures and student-led presentations are provided, as detailed in the Syllabus. The curriculum advances beyond mere application of existing software; it endeavors to impart a thorough understanding of the core principles behind today's cutting-edge techniques, equipping students with the skills to devise and pioneer the forthcoming wave of bioinformatics tools.
</p>
<h2>Prerequisites: </h2>
<p>This course is intended to be a first course in Bioinformatics for both undergraduate and graduate students in Electronic Engineering, Computer Science, Mathematics, Statistics, and Life Sciences. 
Students from non-biological disciplines may wonder about the level of prior biological knowledge required to pursue studies in bioinformatics. The answer is minimal; no advanced understanding beyond what is typically taught in high school biology is necessary. <b>But you will need a willingness to learn biological concepts needed to understand the computational problems we study. We will begin with a quick review of some of the biological concepts needed for bioinformatics, and will learn more as needed.</b><br />
Students are expected to have the following background:
</p>
<ul>
<li><p>Basic programming skills to write a reasonably non-trivial computer program in Python.
</p>
</li>
<li><p>At least an undergraduate course in algorithms (a graduate course in algorithms is preferred).
</p>
</li>
<li><p>Undergraduate or graduate level machine learning courses.
</p>
</li>
<li><p>Students without this background should discuss their preparation with the instructor. The introduction session in the first week of the class will give an overview of the expected background.
</p>
</li>
</ul>
<h2>Optional textbooks</h2>
<p>“Biological sequence analysis” by Durbin, Eddy, Krogh, Mitchison<br />
“An introduction to bioinformatics algorithms” by Jones, Pevzner<br />
“Network Biology:” section in Topology of molecular interaction networks<br />
“Deep Learning” by Ian Goodfellow, Yoshua Bengio, and Aaron Courville<br />
Online book &ldquo;Bioinformatics Algorithms&rdquo; <a href="https://www.bioinformaticsalgorithms.org/bioinformatics-chapter-1" target=&ldquo;blank&rdquo;>[link]</a>,  Phillip Compeau and Pavel Pevzner
</p>
<h2>Grading</h2>
<ul>
<li><p>Homework (50%)
</p>
</li>
<li><p>Final projects (50%)
</p>
</li>
</ul>
<h2>Assignments</h2>
<p>There will be <b>FOUR</b> homework assignments including both theory and programming problems. 
Programming projects can be written in both R and Python. Deep learning final projects should be written in either Tensorflow and PyTorch. 
</p>
<h2>Late policy</h2>
<p>Assignments are to be submitted by the due date listed. Assignments will NOT BE accepted if they are submitted late. Additional extensions will be granted only due to serious and documented medical or family emergencies. 
</p>
<h2>Final projects</h2>
<p>Students are encouraged to work as teams of two or three. Each team can either choose the provided projects on the course website or a separate project related to Biology. If you choose to do your own project, please contact me beforehand to discuss it. Here are the bioinformatics projects you might consider:
</p>
<ul>
<li><p>Easy projects:
</p>
<ul>
<li><p>Using deep neural networks to predict protein secondary structures. 
</p>
</li>
<li><p>Using deep neural networks to predict protein contact/distance map.
</p>
</li>
<li><p>Implement a RNA secondary structure prediction algorithm.
</p>
</li>
<li><p>Implement a single cell analysis pipeline.
</p>
</li>
<li><p>Predicting protein functions prediction using multiple protein interaction networks.
</p>
</li>
<li><p>Implement an algorithm to correct the batch effect in single cell data.
</p>
</li>
<li><p>Predicting enhancers using genomics features.
</p>
</li>
</ul>

</li>
</ul>
<ul>
<li><p>Advanced projects:
</p>
<ul>
<li><p>Compression of protein multiple sequence alignment files.
</p>
</li>
<li><p>Implement a (hierarchical) network community detection algorithms to recover Gene Ontology.
</p>
</li>
<li><p>Protein function prediction from interaction network using graph.				
</p>
</li>
</ul>

</li>
</ul>
<h2>Syllabus (tentative)</h2>
<h3>Sequence Analysis (3 classes)</h3>
<table id="Syllabus">
<tr class="r1"><td class="c1"><b>Time</b> </td><td class="c2" style="text-align:left"> <b>Topic</b> </td><td class="c3" style="text-align:left"> <b>Contents</b> </td></tr>
<tr class="r2"><td class="c1">09/09 </td><td class="c2" style="text-align:left"> Introduction &amp; Sequence alignment</td><td class="c3" style="text-align:left"> (1) Basic biology knowledge; (2) Syllabus and grading policy; (3) Final projects and presentations; (4) Dynamic programming; (5) Local alignment; (6) Global alignment <br /><b>Optional Reading:</b><br />What Is the Role for Algorithmics and Computational Biology in Responding to the COVID-19 Pandemic? <a href="https://www.cell.com/cell-systems/fulltext/S2405-4712(20)30151-4" target=&ldquo;blank&rdquo;>[paper link]</a></td></tr>
<tr class="r5"><td class="c1">
09/23 </td><td class="c2" style="text-align:left"> Homology search (I) </td><td class="c3" style="text-align:left"> (1) Homology; (2) PAM; (3) BLOSUM; (4) Extreme Value Theory; (5) Blast; (6) PSIblast; (7) Multiple sequence alignment </td></tr>
<tr class="r6"><td class="c1">
09/30 </td><td class="c2" style="text-align:left"> Homology search (II) </td><td class="c3" style="text-align:left"> (1) Hidden Markov Models; (2) HMM-based sequence alignment; (3) HMMER; (4) HHpred/HHsearch; (5) HHblits 
</td></tr></table>
<h3>Introduction to Machine Learning (2 classes)</h3>
<table id="Syllabus">
<tr class="r1"><td class="c1"><b>Time</b> </td><td class="c2" style="text-align:left"> <b>Topic</b> </td><td class="c3" style="text-align:left"> <b>Contents</b> </td></tr>
<tr class="r2"><td class="c1">
10/14 </td><td class="c2" style="text-align:left"> Introduction to Machine Learning </td><td class="c3" style="text-align:left"> (1) Introduction (2) AI &amp; Neural Network history; (3) ML task, Overfitting, Underfitting; (4) Linear Regression; (5) MLE; (6) Neural Network; (7) Gradient Descent; (8) Backpropagation algorithm <b>Optional reading:</b><br />(1) Deep Learning in Bioinformatics <a href="https://arxiv.org/pdf/1603.06430.pdf" target=&ldquo;blank&rdquo;>[pdf]</a> <br />(2) Deep learning for computational biology <a href="https://www.embopress.org/doi/pdf/10.15252/msb.20156651" target=&ldquo;blank&rdquo;>[pdf]</a> <br />(3) Machine Learning in Genomic Medicine <a href="https://ieeexplore.ieee.org/stamp/stamp.jsp?reload=true&amp;arnumber=7347331" target=&ldquo;blank&rdquo;>[pdf]</a> <br />(4) Awesome DeepBio <a href="https://github.com/gokceneraslan/awesome-deepbio" target=&ldquo;blank&rdquo;>[github]</a> <br />(5) PyTorch Performance Tuning Guide <a href="https://www.youtube.com/watch?v=9mS1fIYj1So" target=&ldquo;blank&rdquo;>[youtube]</a> <br />(6) The graph neural network model <a href="https://persagen.com/files/misc/scarselli2009graph.pdf" target=&ldquo;blank&rdquo;>[pdf]</a> <br />(7) Graph Representation Learning Book <a href="https://www.cs.mcgill.ca/~wlh/grl_book/" target=&ldquo;blank&rdquo;>[website]</a> </td></tr>
<tr class="r11"><td class="c1">
10/21 </td><td class="c2" style="text-align:left"> Introduction to Deep Learning </td><td class="c3" style="text-align:left"> (1) Convolutional Neural Network; (2) Recurrent Neural Network; (3) Graph Neural Network; (4) Transformer<br /><b>Optional reading:</b><br />(1) Attention Is All You Need <a href="https://arxiv.org/abs/1706.03762" target=&ldquo;blank&rdquo;>[pdf]</a> <br />(2) Transformer from scratch using pytorch <a href="https://www.kaggle.com/code/arunmohan003/transformer-from-scratch-using-pytorch" target=&ldquo;blank&rdquo;>[link]</a> <br />
</td></tr></table>
<h3>Structure Biology (3 classes)</h3>
<table id="Syllabus">
<tr class="r1"><td class="c1"><b>Time</b> </td><td class="c2" style="text-align:left"> <b>Topic</b> </td><td class="c3" style="text-align:left"> <b>Contents</b> </td></tr>
<tr class="r2"><td class="c1"> 10/28 </td><td class="c2" style="text-align:left"> Protein structure prediction </td><td class="c3" style="text-align:left"> (1) Template-based modelling; (2) Co-evolutation; (3) Template-free folding; (4) RFdiffusion &amp; ProteinMPNN; (5) Protein structure alignment<br /><b>Optional Reading:</b><br />(1) Fold proteins by playing games <a href="https://fold.it/" target=&ldquo;blank&rdquo;>[link]</a><br />(2) Fold proteins by deep learning <a href="https://deepmind.com/blog/article/AlphaFold-Using-AI-for-scientific-discovery" target=&ldquo;blank&rdquo;>[link]</a><br /> (3) TM-align: a protein structure alignment algorithm based on the TM-score <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1084323/pdf/gki524.pdf" target=&ldquo;blank&rdquo;>[pdf]</a> <br />(4) Protein structure alignment beyond spatial proximity <a href="https://www.nature.com/articles/srep01448.pdf" target=&ldquo;blank&rdquo;>[pdf]</a> <br />(5) Matt: Local Flexibility Aids Protein Multiple Structure Alignment <a href="https://journals.plos.org/ploscompbiol/article/file?id=10.1371/journal.pcbi.0040010&amp;type=printable" target=&ldquo;blank&rdquo;>[pdf]</a> </td></tr>
<tr class="r9"><td class="c1">
11/04 </td><td class="c2" style="text-align:left"> RNA structure prediction </td><td class="c3" style="text-align:left"> (1) RNA structures and functions; (2) Nussinov Algorithm; (3) Zucker algorithm; (4) Stochastic context free grammars (SCFG)<br /><b>Optional Reading:</b><br />RNA Secondary Structure Prediction By Learning Unrolled Algorithms <a href="https://iclr.cc/virtual_2020/poster_S1eALyrYDH.html" target=&ldquo;blank&rdquo;>[link]</a> </td></tr>
<tr class="r12"><td class="c1">
11/11 </td><td class="c2" style="text-align:left"> AlphaFold </td><td class="c3" style="text-align:left"> (1) MSA transformer; (2) Structural Module; (3) FAPE loss; (4) Triangle Multiplication<br /><b>Optional Reading:</b><br />(1) Improved protein structure prediction using potentials from deep learning <a href="https://www.nature.com/articles/s41586-019-1923-7" target=&ldquo;blank&rdquo;>[pdf]</a><br />(2) Highly accurate protein structure prediction with AlphaFold <a href="https://www.nature.com/articles/s41586-021-03819-2" target=&ldquo;blank&rdquo;>[pdf]</a><br />(3) Accurate structure prediction of biomolecular interactions with AlphaFold 3 <a href="https://www.nature.com/articles/s41586-024-07487-w" target=&ldquo;blank&rdquo;>[pdf]</a>
</td></tr></table>
<h3>AI Drug Development (1 classes)</h3>
<table id="Syllabus">
<tr class="r1"><td class="c1"><b>Time</b> </td><td class="c2" style="text-align:left"> <b>Topic</b> </td><td class="c3" style="text-align:left"> <b>Contents</b> </td></tr>
<tr class="r2"><td class="c1">11/18 </td><td class="c2" style="text-align:left"> CADD &amp; AIDD </td><td class="c3" style="text-align:left"> (1) Small Molecule Drug; (2) Antibody; (3) Mathematical Representation; (3) Structure-based drug design; (4) ADMET prediction; (5) Retrosynthesis algorithm; (6) Drug Repositioning </td></tr>
<tr class="r3"><td class="c1">
</td></tr></table>
<h3>Single Cell in Bioinformatics (2 classes)</h3>
<table id="Syllabus">
<tr class="r1"><td class="c1"><b>Time</b> </td><td class="c2" style="text-align:left"> <b>Topic</b> </td><td class="c3" style="text-align:left"> <b>Contents</b> </td></tr>
<tr class="r2"><td class="c1">11/25 </td><td class="c2" style="text-align:left"> Single cell analysis (I) </td><td class="c3" style="text-align:left"> (1) Intro to single cell technology; (2) Differential expression analysis; (3) Dimension reduction: PCA, TSNE; (4) Missing data imputation: MAGIC, DCA algorithms; (5) Batch effect correction: MNN algorithm<br /><b>Optional reading:</b><br />(1) Single-cell RNA sequencing technologies and bioinformatics pipelines <a href="https://mit6874.github.io/assets/scRNA-pipelines.pdf" target=&ldquo;blank&rdquo;>[pdf]</a><br />(2) Comprehensive integration of single cell data <a href="https://mit6874.github.io/assets/scRNA-integration.pdf" target=&ldquo;blank&rdquo;>[pdf]</a><br />(3) Single-cell RNA sequencing for the study of development, physiology and disease <a href="https://mit6874.github.io/assets/scRNA-disease.pdf" target=&ldquo;blank&rdquo;>[pdf]</a><br />(4) A Systematic Evaluation of Single-cell RNA-sequencing Imputation Methods <a href="https://www.biorxiv.org/content/10.1101/2020.01.29.925974v1.full.pdf" target=&ldquo;blank&rdquo;>[pdf]</a> <br />(5) A benchmark of batch-effect correction methods for single-cell RNA sequencing data <a href="https://genomebiology.biomedcentral.com/track/pdf/10.1186/s13059-019-1850-9" target=&ldquo;blank&rdquo;>[pdf]</a> </td></tr>
<tr class="r9"><td class="c1">12/02 </td><td class="c2" style="text-align:left"> Single cell analysis (II) </td><td class="c3" style="text-align:left"> (1) Trajectory embedding; (2) RNA velocity; (3) Bulk cell deconvolution; (4) Multi-omics integration; (5) Doublets and multiplet detection <br /><b>Optional reading:</b><br />(1) Lineage tracing meets single-cell omics: opportunities and challenges <a href="https://www.nature.com/articles/s41576-020-0223-2" target=&ldquo;blank&rdquo;>[pdf]</a> <br />(2) Sparse Inverse Covariance Estimation with the Graphical Lasso <a href="https://projecteuclid.org/download/pdfview_1/euclid.aos/1152540754" target=&ldquo;blank&rdquo;>[pdf]</a><br />(3) Foldit computer game <a href="https://fold.it/portal/puzzles" target=&ldquo;blank&rdquo;>[link]</a>
</td></tr></table>
<h3>Network in Bioinformatics (2 classes)</h3>
<table id="Syllabus">
<tr class="r1"><td class="c1"><b>Time</b> </td><td class="c2" style="text-align:left"> <b>Topic</b> </td><td class="c3" style="text-align:left"> <b>Contents</b> </td></tr>
<tr class="r2"><td class="c1">12/09 </td><td class="c2" style="text-align:left"> Network motif detection </td><td class="c3" style="text-align:left"> (1) Introduction of Network Biology (2) Basic concepts about graphs (3) Random graphs (4) Network motifs in biological networks (25) G-tries algorithm </td></tr>
<tr class="r3"><td class="c1">
12/16 </td><td class="c2" style="text-align:left"> Network Alignment </td><td class="c3" style="text-align:left"> (1) Network visualization (2) PathBLAST (3) IsoRank (4) Representation-based network alignments <br /><b>Optional Reading:</b><br />(1) REGAL: Representation Learning-based Graph Alignment <a href="https://arxiv.org/pdf/1802.06257.pdf" target=&ldquo;blank&rdquo;>[pdf]</a><br />(2) Deep Adversarial Network Alignment <a href="https://arxiv.org/pdf/1902.10307.pdf" target=&ldquo;blank&rdquo;>[pdf]</a> 
</td></tr></table>
<h3>Project presentations (1 classes)</h3>
<table id="Syllabus">
<tr class="r1"><td class="c1"><b>Time</b> </td><td class="c2" style="text-align:left"> <b>Topic</b> </td><td class="c3" style="text-align:left"> <b>Presentors</b> </td></tr>
<tr class="r2"><td class="c1">12/23 </td><td class="c2" style="text-align:left"> </b></font><font color=red <b> Project presentations </b></font> </td><td class="c3" style="text-align:left"> TBD 
</td></tr></table>
<div id="footer">
<div id="footer-text">
Page generated 2024-08-11 13:25:58 CST, by <a href="https://github.com/wsshin/jemdoc_mathjax" target="blank">jemdoc+MathJax</a>.
(<a href="teaching_intelligence_ba.jemdoc">source</a>)
</div>
</div>
</td>
</tr>
</table>
</body>
</html>
