<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN"
  "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
<head>
<meta name="generator" content="jemdoc, see http://jemdoc.jaboc.net/" />
<meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<link rel="stylesheet" href="jemdoc.css" type="text/css" />
<link rel="stylesheet" href="person.css" type="text/css" />
<title></title>
</head>
<body>
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
try {
    var pageTracker = _gat._getTracker("UA-31107207-1");
    pageTracker._trackPageview();
} catch(err) {}</script>
<table summary="Table for page layout." id="tlayout">
<tr valign="top">
<td id="layout-menu">
<div class="menu-category">Menu</div>
<div class="menu-item"><a href="index.html" class="current">Home</a></div>
<div class="menu-item"><a href="publication.html">Publications</a></div>
<div class="menu-item"><a href="teaching.html">CS590&nbsp;(Graphs&nbsp;in&nbsp;Machine&nbsp;Learning)</a></div>
<div class="menu-item"><a href="teaching_ba.html">CS590&nbsp;(Bioinformatics&nbsp;Algorithms)</a></div>
<div class="menu-item"><a href="software.html">Software</a></div>
</td>
<td id="layout-content">
<h1>CS 59000: Bioinformatics Algorithms (Fall 2020)</h1>
<h2>Course Information</h2>
<p>When: Monday/Wednesday 4:30 pm &ndash; 5:45 pm. <br />
Where: LWSN B134. <br />
Instructor: Jianzhu Ma, majianzhu@purdue.edu.<br />
Office Hour: Thursday. 4:30 pm &ndash; 5:45 pm.or by appointment (send e-mail) Where: TBD<br />
Online discussion is available at Blackboard (mycourses.purdue.edu).</p>
<h2>Course description:</h2>
<p>This course covers almost all the major research directions in bioinformatics and computational biology. It is a type of “all-you-need” class if you want to develop algorithms to process and understand biological data. By completing this course, students will encounter a handful of fundamental algorithmic approaches deriving straight from very widely cited primary literature, much of which has been published in recent years. The course also introduces basic concepts in statistics, mathematics, and machine learning including deep learning, which are all necessary to understand these approaches. Classes consist of both instructor and student presentations of their final projects (see Syllabus). It goes beyond just invoking existing popular programs and packages. Instead, the aim is for you to understand the underlying principles of the most successful techniques that are currently in use, and provide you with the capacity to design and implement the next generation of tools. Another goal of the course is to tackle the research frontiers of computational biology, and give you a glimpse of how research works, expose you to current bioinformatics research directions and guide you to find the problems most interesting to you.</p>
<h2>Prerequisites: </h2>
<p>This course is intended to be a first course in Bioinformatics algorithms for graduate students in computer science, computer engineering, mathematics, statistics, computational life sciences, and related fields. Students from computer science, the engineering disciplines, and mathematical sciences may be concerned about how much previous exposure to biological sciences they need to study bioinformatics. The answer is, not much—students do not need any previous knowledge of biology beyond high school. <b>But you will need a willingness to learn biological concepts needed to understand the computational problems we study. We will begin with a quick review of some of the biological concepts needed for bioinformatics, and will learn more as needed.</b><br />
Students are expected to have the following background:</p>
<ul>
<li><p>Basic programming skills to write a reasonably non-trivial computer program in Python or C (e.g., CS 38003 or equivalent are recommended).</p>
</li>
<li><p>At least an undergraduate course in algorithms (a graduate course in algorithms is preferred).</p>
</li>
<li><p>Undergraduate or graduate level machine learning courses (e.g., CS 37300 and CS 578 are sufficient).</p>
</li>
<li><p>Students without this background should discuss their preparation with the instructor. The introduction session in the first week of the class will give an overview of the expected background.</p>
</li>
</ul>
<h2>Target Students</h2>
<p>This course is intended to be a first course in Bioinformatics algorithms for graduate students in computer science, computer engineering, mathematics, statistics, computational life sciences, and related fields. This is suitable as a foundational course for students planning to do research in bioinformatics with a focus on algorithms; it is also suitable for those who wish to acquire some breadth in algorithms during graduate study and learn about important recent developments in bioinformatics.</p>
<h2>Optional textbooks</h2>
<p>“Biological sequence analysis” by Durbin, Eddy, Krogh, Mitchison<br />
“An introduction to bioinformatics algorithms” by Jones, Pevzner<br />
“Network Biology:” section in Topology of molecular interaction networks<br />
“Deep Learning” by Ian Goodfellow, Yoshua Bengio, and Aaron Courville<br />
Online book &ldquo;Bioinformatics Algorithms&rdquo; <a href="https://www.bioinformaticsalgorithms.org/bioinformatics-chapter-1">[link]</a>,  Phillip Compeau and Pavel Pevzner</p>
<h2>Grading</h2>
<ul>
<li><p>Homework (50%)</p>
</li>
<li><p>Final projects (50%)</p>
</li>
</ul>
<h2>Assignments and exams</h2>
<p>There will be <b>FIVE</b> homework assignments including both theory and programming problems. Assignments should be submitted online in Brightspace, piazza or Blackboard systems. Programming projects can be written in both R and Python. Deep learning final projects should be written in either Tensorflow and PyTorch. </p>
<h2>Late policy</h2>
<p>Assignments are to be submitted by the due date listed. Assignments will NOT BE accepted if they are submitted late. Additional extensions will be granted only due to serious and documented medical or family emergencies. </p>
<h2>Final projects</h2>
<p>Students are encouraged to work as teams of two or three. Each team can either choose the provided projects on the course website or a separate project related to Biology. If you choose to do your own project, please contact me beforehand to discuss it. Here are the bioinformatics projects you might consider:</p>
<ul>
<li><p>Easy projects:</p>
<ul>
<li><p>Using deep neural networks to predict protein secondary structures. </p>
</li>
<li><p>Using deep neural networks to predict protein contact/distance map.</p>
</li>
<li><p>Implement a RNA secondary structure prediction algorithm.</p>
</li>
<li><p>Implement a single cell analysis pipeline.</p>
</li>
<li><p>Predicting protein functions prediction using multiple protein interaction networks.</p>
</li>
<li><p>Implement an algorithm to correct the batch effect in single cell data.</p>
</li>
<li><p>Predicting enhancers using genomics features.</p>
</li></ul>
</li>
<li><p>Advanced projects:</p>
<ul>
<li><p>Implement a (hierarchical) network community detection algorithms to recover Gene Ontology.</p>
</li>
<li><p>Using Contralign algorithm to improve the consistency of multiple network alignment.</p>
</li>
<li><p>Using network and deep learning to predict disease related genes.</p>
</li>
<li><p>Using deep learning to predict drug response for cancer.				</p>
</li>
<li><p>Protein function prediction from interaction network using graph.				</p>
</li>
<li><p>Comparison and analysis of methods in 3D genome reconstruction from Hi-C data. </p>
</li>
<li><p>Translation of genotype to phenotype by using the molecular networks.</p>
</li>
</ul>

</li>
</ul>
<h2>Syllabus (tentative)</h2>
<h3>Sequence analysis in Bioinformatics (9 classes)</h3>
<table id="Syllabus">
<tr class="r1"><td class="c1"><b>Time</b> </td><td class="c2" style="text-align:left"> <b>Topic</b> </td><td class="c3" style="text-align:left"> <b>Contents</b> </td></tr>
<tr class="r2"><td class="c1"> 08/24 </td><td class="c2" style="text-align:left"> Introduction </td><td class="c3" style="text-align:left"> (1) Motivation (2) Basic biology knowledge (3) Syllabus and grading policy (4) Final projects and presentations <br /><b>Optional Reading:</b><br />What Is the Role for Algorithmics and Computational Biology in Responding to the COVID-19 Pandemic? <a href="https://www.cell.com/cell-systems/fulltext/S2405-4712(20)30151-4">[paper link]</a></td></tr>
<tr class="r5"><td class="c1"> 08/26 </td><td class="c2" style="text-align:left"> Dynamic programming and sequence alignment </td><td class="c3" style="text-align:left"> (1) Dynamic programming (2) Local alignment (3) Global alignment </td></tr>
<tr class="r6"><td class="c1">09/02 </td><td class="c2" style="text-align:left"> Homology search I </td><td class="c3" style="text-align:left"> (1) PAM score (2) BLOSUM score (3) Extreme Value Theory (4) Blast </td></tr>
<tr class="r7"><td class="c1">09/07 </td><td class="c2" style="text-align:left"> Multiple sequence alignment </td><td class="c3" style="text-align:left"> (1) Star algorithms (2) CLUSTALW (3) T-coffee (4) M-coffee (5) MUSCLE (6) Probcons </td></tr>
<tr class="r8"><td class="c1">09/09 </td><td class="c2" style="text-align:left"> Advanced sequence alignment algorithms </td><td class="c3" style="text-align:left"> (1) Protein sequence alignment (2) CNFpred (3) co-evolutation alignment (4) ILP-based alignment </td></tr>
<tr class="r9"><td class="c1">09/14 </td><td class="c2" style="text-align:left"> Comparative Genomics </td><td class="c3" style="text-align:left"> (1) Whole genome alignment (2) Suffix tries (3) MUMmer<br /><b>Optional Reading:</b><br />MUMmer 1.0: Alignment of Whole Genomes.<br />MUMmer 2.0: Fast Algorithms for Large-scale Genome Alignment and Comparision.<br />MUMmer 3.0: Versatile and open software for comparing large genomes.<br />MUMmer 4.0: "MUMmer4: A fast and versatile genome alignment system </td></tr>
<tr class="r15"><td class="c1">
09/16 </td><td class="c2" style="text-align:left"> Hidden Markov Models </td><td class="c3" style="text-align:left"> (1) Basic concepts (2) Probability estimation (3) Finding maximum assignment (4) Parameter estimation of HMM </td></tr>
<tr class="r16"><td class="c1">09/21 </td><td class="c2" style="text-align:left"> Homology search II </td><td class="c3" style="text-align:left"> (1) HMM-based sequence alignment (2) HMMER (3) HHpred/HHsearch (4) HHblits (5) ILP-based methods </td></tr>
<tr class="r17"><td class="c1">09/23 </td><td class="c2" style="text-align:left"> Molecular Evolution and Phylogenetics </td><td class="c3" style="text-align:left"> (1) Basics and definitions (2) Jukes Cantor distance (3) UPGMA and Neighbor joining algorithms (4) Sankoff algorithm (5) Fitch algorithm<br /><b>Optional Reading:</b><br />Phylogenetic tree building in the genomic age <a href="https://www.nature.com/articles/s41576-020-0233-0">[review article]</a>
</td></tr></table>
<h3>Structure analysis in Bioinformatics (4 classes)</h3>
<table id="Syllabus">
<tr class="r1"><td class="c1"><b>Time</b> </td><td class="c2" style="text-align:left"> <b>Topic</b> </td><td class="c3" style="text-align:left"> <b>Contents</b> </td></tr>
<tr class="r2"><td class="c1">09/28 </td><td class="c2" style="text-align:left"> Protein structure alignment </td><td class="c3" style="text-align:left"> (1) Basic knowledge of protein structures (2) RMSD (3) SSAP algorithm, (4) DALI algorithm (4) LOCK algorithm (5) Structure databases <br /><b>Optional reading:</b><br />TM-align: a protein structure alignment algorithm based on the TM-score <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1084323/pdf/gki524.pdf">[pdf]</a> <br />Protein structure alignment beyond spatial proximity <a href="https://www.nature.com/articles/srep01448.pdf">[pdf]</a> <br />Matt: Local Flexibility Aids Protein Multiple Structure Alignment <a href="https://journals.plos.org/ploscompbiol/article/file?id=10.1371/journal.pcbi.0040010&amp;type=printable">[pdf]</a></td></tr>
<tr class="r7"><td class="c1">
09/30 </td><td class="c2" style="text-align:left"> Protein structure prediction </td><td class="c3" style="text-align:left"> (1) Template-based modelling (2) Raptor (3) Template-free folding (4) Rosetta<br /><b>Optional Reading:</b><br />(1) Fold proteins by playing games <a href="https://fold.it/">[link]</a><br />(2) Fold proteins by deep learning <a href="https://deepmind.com/blog/article/AlphaFold-Using-AI-for-scientific-discovery">[link]</a> </td></tr>
<tr class="r11"><td class="c1">
10/05 </td><td class="c2" style="text-align:left"> RNA structure prediction </td><td class="c3" style="text-align:left"> (1) RNA structures and functions (2) Nussinov Algorithm (3) Zucker algorithm (4) Stochastic context free grammars (SCFG)<br /><b>Optional Reading:</b><br />RNA Secondary Structure Prediction By Learning Unrolled Algorithms <a href="https://iclr.cc/virtual_2020/poster_S1eALyrYDH.html">[link]</a> </td></tr>
<tr class="r14"><td class="c1">
10/07 </td><td class="c2" style="text-align:left"> 3D genome analysis </td><td class="c3" style="text-align:left"> (1) Basic knowledge about HiC techniques (2) Algorithms to detect Topologically Associating Domains (TAD) (3) Algorithms to Predict Enhancer-Promoter interactions (PEP)<br /><b>Optional reading:</b><br />1. Comprehensive Mapping of Long-Range Interactions Reveals Folding Principles of the Human Genome <a href="https://science.sciencemag.org/content/326/5950/289">[pdf]</a> <br />2. Topological Domains in Mammalian Genomes Identified by Analysis of Chromatin Interactions <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3356448/pdf/nihms366885.pdf">[pdf]</a> <br />3. A 3D Map of the Human Genome at Kilobase Resolution Reveals Principles of Chromatin Looping <a href="https://www.cell.com/action/showPdf?pii=S0092-8674%2814%2901497-4">[pdf]</a> <br />4. Single-cell Hi-C reveals cell-to-cell variability in chromosome structure <a href="https://www.nature.com/articles/nature12593.pdf">[pdf]</a> <br />5. 3D structures of individual mammalian genomes studied by single-cell Hi-C <a href="https://www.nature.com/articles/nature21429.pdf">[pdf]</a> 
</td></tr></table>
<h3>Machine Learning in Bioinformatics (5 classes)</h3>
<table id="Syllabus">
<tr class="r1"><td class="c1"><b>Time</b> </td><td class="c2" style="text-align:left"> <b>Topic</b> </td><td class="c3" style="text-align:left"> <b>Contents</b> </td></tr>
<tr class="r2"><td class="c1">10/12 </td><td class="c2" style="text-align:left"> Research talks </td><td class="c3" style="text-align:left"> Given by Prof. <a href="http://kazemianlab.com/">Majid Kazemian</a>, Prof. <a href="http://kiharalab.org/">Daisuke Kihara</a> and Prof. <a href="https://www.cs.purdue.edu/homes/ma634/">Jianzhu Ma</a> </td></tr>
<tr class="r3"><td class="c1">
10/14 </td><td class="c2" style="text-align:left"> Introduction to Deep Learning </td><td class="c3" style="text-align:left"> (1) Basic concepts, MLP (2) Convolutional Neural Network (3) Recurrent Neural Network (4) Long Short Term Memory (5) Resnet<br /><b>Optional reading:</b><br />(1) Deep Learning in Bioinformatics <a href="https://arxiv.org/pdf/1603.06430.pdf">[pdf]</a> <br />(2) Deep learning for computational biology <a href="https://www.embopress.org/doi/pdf/10.15252/msb.20156651">[pdf]</a> <br />(3) Machine Learning in Genomic Medicine <a href="https://ieeexplore.ieee.org/stamp/stamp.jsp?reload=true&amp;arnumber=7347331">[pdf]</a> <br />(4) Awesome DeepBio <a href="https://github.com/gokceneraslan/awesome-deepbio">[github]</a> <br />(5) PyTorch Performance Tuning Guide <a href="https://www.youtube.com/watch?v=9mS1fIYj1So">[youtube]</a> </td></tr>
<tr class="r10"><td class="c1">
10/19 </td><td class="c2" style="text-align:left"> Deep Learning in genomics &amp; protein structures </td><td class="c3" style="text-align:left"> (1) 1D CNN and DeepSea (2) Point cloud modelling and 3D CNN (3) Model interpretations <b>Optional reading:</b><br />(1) Adaptive Deconvolutional Networks for Mid and High Level Feature Learning <a href="https://ieeexplore.ieee.org/document/6126474">[pdf]</a> <br />(2) Deep Inside Convolutional Networks: Visualising Image Classification Models and Saliency Maps <a href="https://arxiv.org/pdf/1312.6034.pdf">[pdf]</a> <br />(3) The Mythos of Model Interpretability <a href="https://arxiv.org/pdf/1606.03490.pdf">[pdf]</a></td></tr>
<tr class="r15"><td class="c1">
10/21 </td><td class="c2" style="text-align:left"> Deep Learning on biological networks </td><td class="c3" style="text-align:left"> (1) GraphSage (2) Graph Convolutional Neural Network (3) Graph Attention Models (4) Popular datasets<br /><b>Optional reading:</b> <br />(1) The graph neural network model <a href="https://persagen.com/files/misc/scarselli2009graph.pdf">[pdf]</a> <br />(2) Gated Graph Sequence Neural Networks <a href="https://arxiv.org/pdf/1511.05493.pdf">[pdf]</a> <br />(3) Graph Representation Learning Book <a href="https://www.cs.mcgill.ca/~wlh/grl_book/">[website]</a> </td></tr>
<tr class="r20"><td class="c1">
10/26 </td><td class="c2" style="text-align:left"> Deep Learning on small moleculars </td><td class="c3" style="text-align:left"> (1) (2) (3) 
</td></tr></table>
<h3>Matrix analysis in Bioinformatics (4 classes)</h3>
<table id="Syllabus">
<tr class="r1"><td class="c1"><b>Time</b> </td><td class="c2" style="text-align:left"> <b>Topic</b> </td><td class="c3" style="text-align:left"> <b>Contents</b> </td></tr>
<tr class="r2"><td class="c1">
10/28 </td><td class="c2" style="text-align:left"> Gene expression analysis </td><td class="c3" style="text-align:left"> (1) Differential expression analysis (2) Dimension reduction: PCA, TSNE </td></tr>
<tr class="r3"><td class="c1">
11/02 </td><td class="c2" style="text-align:left"> Single cell analysis I </td><td class="c3" style="text-align:left"> (1) Single cell techniques (2) Missing data imputation: MAGIC, DCA algorithms (3) Batch effect correction: MNN algorithm<br /><b>Optional reading:</b><br />(1) Single-cell RNA sequencing technologies and bioinformatics pipelines <a href="https://mit6874.github.io/assets/scRNA-pipelines.pdf">[pdf]</a><br />(2) Comprehensive integration of single cell data <a href="https://mit6874.github.io/assets/scRNA-integration.pdf">[pdf]</a><br />(3) Single-cell RNA sequencing for the study of development, physiology and disease <a href="https://mit6874.github.io/assets/scRNA-disease.pdf">[pdf]</a><br />(4) A Systematic Evaluation of Single-cell RNA-sequencing Imputation Methods <a href="https://www.biorxiv.org/content/10.1101/2020.01.29.925974v1.full.pdf">[pdf]</a> <br />(5) A benchmark of batch-effect correction methods for single-cell RNA sequencing data <a href="https://genomebiology.biomedcentral.com/track/pdf/10.1186/s13059-019-1850-9">[pdf]</a> </td></tr>
<tr class="r10"><td class="c1">
11/04 </td><td class="c2" style="text-align:left"> Single cell analysis II </td><td class="c3" style="text-align:left"> (1) Trajectory embedding (2) RNA velocity (3) Bulk cell deconvolution <br /><b>Optional reading:</b><br />(1) Lineage tracing meets single-cell omics: opportunities and challenges <a href="https://www.nature.com/articles/s41576-020-0223-2">[pdf]</a> </td></tr>
<tr class="r13"><td class="c1">
11/09 </td><td class="c2" style="text-align:left"> Single cell analysis III </td><td class="c3" style="text-align:left"> (1) Single cell multi-omics: single cell epigenetics, single cell HiC (2) Doublets and multiplets <br /><b>Optional reading:</b><br />(1) Sparse Inverse Covariance Estimation with the Graphical Lasso <a href="https://projecteuclid.org/download/pdfview_1/euclid.aos/1152540754">[pdf]</a><br />(5) Foldit computer game <a href="https://fold.it/portal/puzzles">[link]</a>
</td></tr></table>
<h3>Network analysis in Bioinformatics (4 classes)</h3>
<table id="Syllabus">
<tr class="r1"><td class="c1"><b>Time</b> </td><td class="c2" style="text-align:left"> <b>Topic</b> </td><td class="c3" style="text-align:left"> <b>Contents</b> </td></tr>
<tr class="r2"><td class="c1">11/11 </td><td class="c2" style="text-align:left"> Network motif detection </td><td class="c3" style="text-align:left"> (1) Introduction of Network Biology (2) Basic concepts about graphs (3) Random graphs (4) Network motifs in biological networks (25) G-tries algorithm </td></tr>
<tr class="r3"><td class="c1">
11/16 </td><td class="c2" style="text-align:left"> Network Alignment </td><td class="c3" style="text-align:left"> (1) Network visualization (2) PathBLAST (3) IsoRank (4) Representation-based network alignments <br /><b>Optional Reading:</b><br />(1) REGAL: Representation Learning-based Graph Alignment <a href="https://arxiv.org/pdf/1802.06257.pdf">[pdf]</a><br />(2) Deep Adversarial Network Alignment <a href="https://arxiv.org/pdf/1902.10307.pdf">[pdf]</a> </td></tr>
<tr class="r7"><td class="c1">
11/18 </td><td class="c2" style="text-align:left"> Graphical Models I </td><td class="c3" style="text-align:left"> (1) General concepts (2) Exact inference  (3) Sum-product algorithm (4) Max-product algorithm (5) Conditional Random Fields<br /><b>Optional textbooks:</b><br />(1) &ldquo;Probabilistic Graphical Models&rdquo; by By Daphne Koller and Nir Friedman <br />(2) &ldquo;Graphical Models, Exponential Families, and Variational Inference&rdquo; by Martin J. Wainwright and Michael I. Jordan <br />(3) Chapter 8 of &ldquo;Pattern Recognition and Machine Learning&rdquo; by Christopher M. Bishop </td></tr>
<tr class="r12"><td class="c1">
11/23 </td><td class="c2" style="text-align:left"> Graphical Models II </td><td class="c3" style="text-align:left"> (1) Structure learning (2) Gaussian Graphical Model (3) Pseudo-likelihood approximation (4) Protein contact prediction (5) Deep-Learning-based structure learning <br /><b>Optional reading:</b><br />(1) Sparse Inverse Covariance Estimation with the Graphical Lasso <a href="https://projecteuclid.org/download/pdfview_1/euclid.aos/1152540754">[pdf]</a><br />(2) High-Dimensional Graphs and Variable Selection with the Lasso <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3019769/pdf/kxm045.pdf">[pdf]</a><br />(3) Protein 3D Structure Computed from Evolutionary Sequence Variation <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0028766">[pdf]</a> <br />(4) AlphaFold: Using AI for scientific discovery <a href="https://deepmind.com/blog/article/alphafold">[BLOG POST]</a> <br />(5) Foldit computer game <a href="https://fold.it/portal/puzzles">[link]</a>
</td></tr></table>
<h3>Project presentations (2 classes)</h3>
<table id="Syllabus">
<tr class="r1"><td class="c1"><b>Time</b> </td><td class="c2" style="text-align:left"> <b>Topic</b> </td><td class="c3" style="text-align:left"> <b>Presentors</b> </td></tr>
<tr class="r2"><td class="c1">11/30 </td><td class="c2" style="text-align:left"> </b></font><font color=red <b> Project presentations </b></font> </td><td class="c3" style="text-align:left"> TBD </td></tr>
<tr class="r3"><td class="c1">12/02 </td><td class="c2" style="text-align:left"> </b></font><font color=red <b> Project presentations </b></font> </td><td class="c3" style="text-align:left"> TBD 
</td></tr></table>
<div id="footer">
<div id="footer-text">
Page generated 2020-09-02 10:53:29 EDT, by <a href="http://jemdoc.jaboc.net/">jemdoc</a>.
(<a href="teaching_ba.jemdoc">source</a>)
</div>
</div>
</td>
</tr>
</table>
</body>
</html>
